Slug generation (#2755)
buildkite/docs: pages/apis/graphql/cookbooks/pipelines.md

> To avoid writing an entire YAML file in a single string, you can place a <code>pipeline.yml</code> file in a <code>.buildkite</code> directory at the root of your repo, and use the <code>pipeline upload</code> command in your pipeline steps to tell Buildkite where to find it. This means you only need the following:
> <code>steps: { yaml: "steps:\n - command: \"buildkite-agent pipeline upload\"" }</code>

> ### Slug creation conventions

Reviewer: This section heading doesn't really explain the purpose of the contents contained within it. Would it be better modifying this to:

```suggestion
### Deriving a pipeline slug from the pipeline's name
```

> Pipeline slugs are generated based on the pipeline name you provide during the pipeline creation.

Reviewer: As an intro sentence, I suggest this wording simplification here, since I suspect in the next couple of sentences, you're going to explain how the slug is based on pipeline name...

```suggestion
Pipeline slugs are generated from the pipeline name you provide during the pipeline creation.
```

The maximum allowed character length for a pipeline slug is `100`.
The supported character format for slug generation is:

> `/\A[a-zA-Z0-9]+[a-zA-Z0-9\-]*\z/`.

Reviewer: This looks like a regex. If so, what is the purpose of mentioning it here?

Can this regex be used to convert a pipeline name to its intended slug? If so, then this needs to be explained here, as this is not obvious.

Reviewer: I'm assuming that this is, based on the reason for including it here.

> All the whitespace characters that appear in the pipeline name consecutively will be converted to a single `-` character (so `"Hello[space]there"` and `"Hello[space, space, space, etc.]there"` will be equally converted to a `hello-there` slug), and uppercase will be converted to lowercase.

Reviewer: Modifying slightly to explain/tie this section of text in with its purpose...

```suggestion
When a pipeline slug is derived from a pipeline's name, all space characters (including consecutive ones) in the pipeline's name are converted to single hyphen `-` characters, and all uppercase characters are converted to their lowercase counterparts. Therefore, pipeline names of either `Hello there friend` or `Hello    there friend` are converted to the slug `hello-there-friend`).
```

> An attempt at creating a new pipeline with a name that matches an existing pipeline name will throw an error.

Reviewer: Minor wording suggestions...

```suggestion
Any attempt to create a new pipeline with a name that matches an existing pipeline's name will result in an error.
```

Is it definitely the pipeline name matching that causes this error, or is it the (derived) pipeline slug matching that causes the error here?
